### PR TITLE
PP-4451 Implements logic for updating Organisation name

### PR DESCRIPTION
**Pull request by cobainc0**
## WHAT
 - add PATCH request to allow selfservice to update merchant name in adminusers
 - updates cypress tests to incorporate new behaviour
 - minor refactoring

solo: @cobainc0

### CODE REVIEW

File: app/controllers/request-to-go-live/organisation-name/post.controller.js [hunk 1]
```diff
@@ -5,27 +5,36 @@ const lodash = require('lodash')
...
   if (lodash.isEmpty(errors)) {
-    // TODO: handle submission
-    res.redirect(
-      303,
-      goLiveStageToNextPagePath[req.service.currentGoLiveStage].replace(':externalServiceId', req.service.externalId)
-    )
+    return updateMerchantName(req.service.externalId, organisationName, req.correlationId)
+      .then(service => {
+        return updateCurrentGoLiveStage(service.externalId, goLiveStage.ENTERED_ORGANISATION_NAME, req.correlationId)
+          .then(updatedService => {
```
Comment (sfount) [inline app/controllers/request-to-go-live/organisation-name/post.controller.js]: This promise structure is nested in the same way a callback chain might be, this code should actually work but it might become hard to follow in the future, especially for error handling. 

The goal should be to only ever have a depth of 1 in promise chains, that way the error is handled uniformly at the bottom and there aren't small issues with promises not being returned further down the chain. 

(we also don't need to return anything to express route handlers)

This would look something like: 
```js 
updateMerchantName()
  .then((service) => { 
    return updateCurrentGoLiveStage()
  })
  .then((updatedService) => { 
    // service successfully updated, redirect user to next stage
    res.redirect()
  })
  .catch((error) => { 
    renderErrorView()
  })
```

File: app/controllers/request-to-go-live/organisation-name/post.controller.js [hunk 2]
```diff
@@ -5,27 +5,36 @@ const lodash = require('lodash')
...
+      })
+      .catch(err => {
+        renderErrorView(req, res, err.message)
+      })
   } else {
     lodash.set(req, 'session.pageData.requestToGoLive.organisationName', {
       success: false,
       errors: errors,
-      organisationName: req.body[REQUEST_ORGANISATION_NAME_FIELD]
+      organisationName: req.body[ORGANISATION_NAME_FIELD]
```
Comment (sfount) [inline app/controllers/request-to-go-live/organisation-name/post.controller.js]: This constant makes the code very clear! 👍

Comment (cobainc0) [reply app/controllers/request-to-go-live/organisation-name/post.controller.js]: Gracias

File: app/controllers/request-to-go-live/organisation-name/post.controller.js (hunk 1 above)
Comment (cobainc0) [reply app/controllers/request-to-go-live/organisation-name/post.controller.js]: Awesome thanks!  I've updated this now 👌🏿
